pandas/core/generic.py

```diff
@@ -915,6 +915,11 @@ def to_hdf(self, path_or_buf, key, **kwargs):
         """
 
         from pandas.io import pytables
+        if isinstance(self, pd.Series) and any(np.iscomplex(self)):
```

Comment:
not the right place for this. Checking is done by the particular format (table/fixed) inside the object itself.